internal/backend/sftp/sftp.go

```diff
@@ -318,7 +332,12 @@ func (r *SFTP) Save(ctx context.Context, h restic.Handle, rd restic.RewindReader
 	}
 
 	err = f.Close()
-	return errors.Wrap(err, "Close")
+	if err != nil {
+		return errors.Wrap(err, "Close")
+	}
+
+	err = r.c.Rename(tmpFilename, filename)
```

Comment:
Rename in SFTP fails when the target file exists. Maybe use PosixRename?

```suggestion
	// PosixRename overwrites an existing directory entry (such as a broken file), but isn't always available.
	if r.c.HasExtension("posix-rename@openssh.com") {
		err = r.c.PosixRename(tmpFilename, filename)
	} else {
		err = r.c.Rename(tmpFilename, filename)
	}
```

Reply:
Hmm, I'm not completely sure its worth the trouble. File collisions should be rather rare as two independent restic processes will likely never end up uploading the same file. And after a failed upload the `RetryBackend` first tries to delete the file and then upload the file again.

Reply:
Fair enough.